### ISSUE TITLE
@abstractstaticmethod and @abstractclassmethod isn't treated as @staticmethod and @classmethod respectively

### ISSUE THREAD

**Issue by Fronkan** [microsoft/pyright#2798]
**Describe the bug**
I am using Pylance and it wrongly reports errors when using functions marked `@abstractclassmethods` or `@abstractstaticmethod`, 
as they are not treated as classmethods and staticmethods, respectively.

**To Reproduce**
For `@abstractclassmethod`:
```python
from abc import abstractclassmethod
class T:
    @abstractclassmethod
    def test(cls):
        pass

    @classmethod
    def test2(cls):
        cls.test()
```
In Pylance `cls.test()` will show the error `Argument missing for parameter "cls"`, indicating it believes this to be the "self" parameter implicitly passed only when using an instance. Just using `@classmethod` instead of `@abstractclassmethod` removes the error.

For `@abstractstaticmethod`:
```python
from abc import abstractstaticmethod
class T:
    @abstractstaticmethod
    def test(name: str):
        pass
```
Here we see the error on the str type-hint. The error is: `Type of parameter "name" must be a supertype of its class "T" `. Again swaping to the non-abstract version (`@staticmethod`) removes the error. It seems to interpret name as the self parameter.

**Expected behavior**
I would expect that `@abstractclassmethod` and `@abstractstaticmethod` would behave like their non-abstract counterparts.

**VS Code extension or command-line**
I am using it through Pylance version v2021.12.2. However, as Pylance uses pyright I figured this issue where more at home here. If this isn't correct, please let me now.

**Comment by erictraut** [comment]
`abstractclassmethod` and `abstractstaticmethod` have been deprecated since Python 3.3, and they are not recommended for use. You should use a chained decorator instead, as per the [Python documentation](https://docs.python.org/3/library/abc.html).

```python
@classmethod
@abstractmethod
def test(cls):
        pass
```

**Comment by Fronkan** [comment]
Oh, I missed that they where deprecated.
Thanks for the response.
